server/api/src/main/java/io/smallrye/graphql/api/Context.java

```diff
@@ -0,0 +1,248 @@
+            return Optional.of(getRequest().getString(OPERATION_NAME));
+        }
+        return Optional.empty();
+    }
+
+    /**
+     * Check if the request contains an operation name
+     * 
+     * @return true if it does
+     */
+    default boolean hasOperationName() {
+        return getRequest().containsKey(OPERATION_NAME)
+                && getRequest().get(OPERATION_NAME) != null
+                && !getRequest().get(OPERATION_NAME).getValueType().equals(JsonValue.ValueType.NULL);
+    }
+
+    /**
+     * Get the variables of the request
+     * Could be null (not set)
+     * 
+     * @return
+     */
+    default Optional<Map<String, Object>> getVariables() {
+        if (hasVariables()) {
+            JsonValue jsonValue = getRequest().get(VARIABLES);
+            return VariablesParser.toMap(jsonValue);
+        }
+        return Optional.empty();
+    }
+
+    /**
+     * Check if the request contains variables
+     * 
+     * @return true if it does
+     */
+    default boolean hasVariables() {
+        return (getRequest().containsKey(VARIABLES)
+                && getRequest().get(VARIABLES) != null
+                && !getRequest().get(VARIABLES).getValueType().equals(JsonValue.ValueType.NULL));
+    }
+
+    /**
+     * Get the execution ID.
+     * 
+     * @return the ID as a String
+     */
+    public String getExecutionId();
+
+    /**
+     * Return true if the argument exist
+     * 
+     * @param name the argument name
+     * @return true if there
+     */
+    public boolean hasArgument(String name);
+
+    /**
+     * Get the argument using a name
+     * This return the argument instance if it exists
+     */
+    public <T> T getArgument(String name);
+
+    /**
+     * Same as above but with the option to do a default value
+     * 
+     * @param <T>
+     * @param name
+     * @param defaultValue
+     * @return the argument instance if it exists, else the provided default
+     */
+    default <T> T getArgumentOrDefault(String name, T defaultValue) {
+        T t = getArgument(name);
+        if (t == null)
+            return defaultValue;
+        return t;
+    }
+
+    /**
+     * Get all the arguments
+     * 
+     * @return a map with name and instance of the argument
+     */
+    public Map<String, Object> getArguments();
+
+    default boolean hasSource() {
+        Object o = getSource();
+        return o != null;
+    }
+
+    public <T> T getSource();
+
+    /**
+     * Return the current path
+     * 
+     * @return the path as a String
+     */
+    public String getPath();
+
+    /**
+     * Return the fields selected in the request
+     * 
+     * @return JsonArray of fields selected
+     */
+    public JsonArray getSelectedFields();
+
+    /**
+     * This leaky abstraction allows falling down to the underlying implementation
+     * 
+     * @param <T> the implementation class
+     * @return instance of the implementation class
+     */
+    public <T> T unwrap();
```

Comment:
I think I would allow this method to pass in the Class type - so that we have more flexibility in the future to allow other types - i.e. 
```suggestion
    public <T> T unwrap(Class<T> wrappedType);
```

Then the method should probably throw an `IllegalArgumentException` for any passed-in type that we can't handle - so for now, anything other than `DataFetchingEnvironment`, but it allows us to add more context types in the future.

Reply:
Sounds good. Thanks. I think this is what @t1 also wanted.